statistics/statistics.go

```diff
@@ -146,27 +140,28 @@ func (p *Plugin) calculateStats(buff interface{}, startTime time.Time, stopTime
 			val, err = stats.Sum(buffer)
 		case "Trimean":
 			val, err = stats.Trimean(buffer)
+		case "Quartile_Range":
+			val, err = stats.InterQuartileRange(buffer)
 		default:
 			st := fmt.Sprintf("Unknown statistic received %T:", stat)
-			log.Warnf(st)
+			log.Errorf(st)
 			err = errors.New(st)
 		}
 
 		if err != nil {
 			log.Warnf("Error in %T", stat)
 		}
 
-		metric := plugin.MetricType{
-			Data_:               val,
-			Namespace_:          core.NewNamespace(namespace, stat),
-			Timestamp_:          time,
-			LastAdvertisedTime_: time,
-			Unit_:               unit,
-			Tags_:               tags,
+		metric := plugin.Metric{
+			Data:      val,
+			Namespace: plugin.NewNamespace(namespace, stat),
+			Timestamp: time,
+			Unit:      unit,
+			Tags:      tags,
 		}
 
 		if stat == "Mode" {
-			metric.Data_ = modeVal
+			metric.Data = modeVal
```

Comment:
returning an array without specifying what the values are. It seems not much use. Do you like me to open an issue for this so that you can work on it later?